subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -187,7 +190,7 @@ public boolean isStarted() {
 
         @Override
         public String toString() {
-            return "Worker ".concat(worker.getDisplayName()).concat(" for operation ").concat(operation.getDescription());
+            return "Worker ".concat(worker.getDisplayName()).concat(" for ").concat(operation.getClass().getSimpleName());
```

Comment:
The problem is that the description has to be build first (`operation.description().build()`). I did not want to do that just for the sake of `toString()`.

I changed it to build it once (if `toString` is called) and cache it in the Holder.